docs/docs/develop/policies.md

```diff
@@ -0,0 +1,118 @@
+
+### User Attributes
+Following user attributes are available for defining policies:
+- `name` - name of the user. Example - `John Doe`    
+- `email` - email of the user. Example - `john.doe@test.com`
+- `domain` - domain of the email. Example - `test.com`
+- `groups` - _**list**_ of groups the user belongs to. Currently, we don't support custom user groups, so the only group available is `all`. Example - `all`
+- `admin` - whether the user is an admin or not. Example - `true`
+
+User attributes will be namespaced with `.user` prefix while being used, more in the example below
+ 
+## Examples
+
+1. Let's say you want to restrict access to a dashboard only to admin users. You can define the following policy in the `dashboard_name.yaml` file under `policy` section.
+    ```yaml 
+    policy:
+      has_access: "'{{ .user.admin }}' == true"
+    ```
+   > **_Note:_**  If `policy` section is defined and `has_access` is not, then it will resolve to `false` meaning dashboard won't be accessible to anyone.
+2. Along with the above policy, you also want to filter all counts/measures as per the user domain. Assuming you have a `domain` dimension in your model. You can define the following policy in the `dashboard_name.yaml` file under `policy` section.
+    ```yaml
+    policy:
+      has_access: "'{{ .user.admin }}' == true"
+      filter: "domain = '{{ .user.domain }}'"
+     ```
+   > **_Note:_**  `filter` condition needs to be a valid SQL `WHERE` clause. It will be injected in the queries after resolving the user attributes templating.
+3. Along with the above policy, you want to exclude certain dimension for user not having `test.com` domain. Assuming you have a `ssn` dimension in your model. You can define the following policy in the `dashboard_name.yaml` file under `policy` section.
+    ```yaml
+    policy:
+      has_access: "'{{ .user.admin }}' == true"
+      filter: "domain = '{{ .user.domain }}'"
+      exclude:
+        - name: ssn
+          if: "{{ .user.domain }} != 'test.com'"
+    ```
+   As per above ssn dimension will be excluded but all other dimensions will be included.
+4. Here's one example for column inclusion.
+
+    ```yaml
+    policy:
+      has_access: "true"
+      filter: "domain = '{{ .user.domain }}'"
+      include:
+        - name: ssn
+          if: "'{{ .user.admin }}' == true"
+    ```
+   > **_Note:_** If include is defined all other dimensions will be excluded. Only one of the
+     `include`/`exclude` can be defined at a time.
+
+> **_Note:_** `has_access` and `if` conditions on `include`/`exclude` fields needs to be valid expressions that evaluates to true/false.
+
+## Advanced Examples
+Policies are resolved in two phases - 
+1. The templating engine replaces placeholder like this `{{ }}` with actual values
+2. For `has_access` and `if` conditions, the resultant expression is evaluated to true/false, whereas for `filter` condition the resultant expression is injected in the query.
```

Comment:
Clarify that it is injected as a SQL expression in the WHERE clause or the underlying query